core/src/main/java/org/hisp/dhis/android/core/dataset/internal/DataSetCompleteRegistrationPostCall.kt

```diff
@@ -136,19 +131,13 @@ internal class DataSetCompleteRegistrationPostCall(
     }
 
     private suspend fun postCompleteRegistrations(
-        payload: DataSetCompleteRegistrationPayload,
+        dataSetCompleteRegistrations: List<DataSetCompleteRegistration>,
     ): Result<DataValueImportSummary, D2Error> {
-        return if (versionManager.isGreaterOrEqualThan(DHISVersion.V2_38)) {
-            coroutineAPICallExecutor.wrap(
-                acceptedErrorCodes = listOf(HttpURLConnection.HTTP_CONFLICT),
-                errorClass = DataValueImportSummaryWebResponse::class.java,
-            ) {
-                dataSetCompleteRegistrationService.postDataSetCompleteRegistrationsWebResponse(payload)
-            }.map { it.response }
-        } else {
-            coroutineAPICallExecutor.wrap {
-                dataSetCompleteRegistrationService.postDataSetCompleteRegistrations(payload)
-            }
+        return coroutineAPICallExecutor.wrap(
+            acceptedErrorCodes = listOf(HttpURLConnection.HTTP_CONFLICT),
+            errorClass = DataValueImportSummary::class.java,
```

Comment:
This class changed from DataValueImportSummaryWebResponse to DataValueImportSummary. And it is not a DTO but a Jackson-annotated class. I think it will work because this class is not deserialized by Ktor but using the ObjectMapper in the error catcher. No need to change it now, but I think we have to think about this cases, maybe moving this part of the logic to the network layer

Reply:
I have just seen the Jackson annotations for this class were removed, so it could failed to be deserialized.